### PR TITLE
[pyright] [core] _core/definitions/asset_reconciliation_sensor

### PR DESCRIPTION
**Pull request by smackesey**
### Summary & Motivation

Slight adjustment to `asset_reconciliation_sensor` to solve type error.

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #11002** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11002" target="_blank">Graphite</a> 
    * **PR #11330** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11330" target="_blank">Graphite</a> 
    * **PR #11001** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11001" target="_blank">Graphite</a> 
    * **PR #11038** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11038" target="_blank">Graphite</a> 
      * **PR #11051** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11051" target="_blank">Graphite</a> 
      * **PR #11362** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11362" target="_blank">Graphite</a> 
        * **PR #11717** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11717" target="_blank">Graphite</a>  👈
          * **PR #11695** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11695" target="_blank">Graphite</a> 
            * **PR #11696** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11696" target="_blank">Graphite</a> 
              * **PR #11697** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11697" target="_blank">Graphite</a> 
                * **PR #11698** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11698" target="_blank">Graphite</a> 
                  * **PR #11699** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11699" target="_blank">Graphite</a> 
                    * **PR #11363** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11363" target="_blank">Graphite</a> 
                      * **PR #11364** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11364" target="_blank">Graphite</a> 
                        * **PR #11365** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11365" target="_blank">Graphite</a> 
                          * **PR #11366** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11366" target="_blank">Graphite</a> 
                            * **PR #11367** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11367" target="_blank">Graphite</a> 
                              * **PR #11368** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11368" target="_blank">Graphite</a> 
                                * **PR #11369** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11369" target="_blank">Graphite</a> 
                                  * **PR #10983** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10983" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/11717?utm_source=stack-comment).

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #11002** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11002" target="_blank">Graphite</a> 
    * **PR #11330** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11330" target="_blank">Graphite</a> 
    * **PR #11001** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11001" target="_blank">Graphite</a> 
    * **PR #11038** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11038" target="_blank">Graphite</a> 
      * **PR #11051** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11051" target="_blank">Graphite</a> 
      * **PR #11362** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11362" target="_blank">Graphite</a> 
        * **PR #11717** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11717" target="_blank">Graphite</a>  👈
          * **PR #11695** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11695" target="_blank">Graphite</a> 
            * **PR #11696** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11696" target="_blank">Graphite</a> 
              * **PR #11697** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11697" target="_blank">Graphite</a> 
                * **PR #11698** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11698" target="_blank">Graphite</a> 
                  * **PR #11699** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11699" target="_blank">Graphite</a> 
                    * **PR #11363** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11363" target="_blank">Graphite</a> 
                      * **PR #11364** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11364" target="_blank">Graphite</a> 
                        * **PR #11365** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11365" target="_blank">Graphite</a> 
                          * **PR #11366** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11366" target="_blank">Graphite</a> 
                            * **PR #11367** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11367" target="_blank">Graphite</a> 
                              * **PR #11368** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11368" target="_blank">Graphite</a> 
                                * **PR #11369** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11369" target="_blank">Graphite</a> 
                                  * **PR #10983** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10983" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/11717?utm_source=stack-comment).

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #11002** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11002" target="_blank">Graphite</a> 
    * **PR #11330** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11330" target="_blank">Graphite</a> 
    * **PR #11001** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11001" target="_blank">Graphite</a> 
    * **PR #11038** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11038" target="_blank">Graphite</a> 
      * **PR #11051** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11051" target="_blank">Graphite</a> 
      * **PR #11362** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11362" target="_blank">Graphite</a> 
        * **PR #11717** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11717" target="_blank">Graphite</a>  👈
          * **PR #11695** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11695" target="_blank">Graphite</a> 
            * **PR #11696** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11696" target="_blank">Graphite</a> 
              * **PR #11697** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11697" target="_blank">Graphite</a> 
                * **PR #11698** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11698" target="_blank">Graphite</a> 
                  * **PR #11699** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11699" target="_blank">Graphite</a> 
                    * **PR #11363** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11363" target="_blank">Graphite</a> 
                      * **PR #11364** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11364" target="_blank">Graphite</a> 
                        * **PR #11365** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11365" target="_blank">Graphite</a> 
                          * **PR #11366** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11366" target="_blank">Graphite</a> 
                            * **PR #11367** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11367" target="_blank">Graphite</a> 
                              * **PR #11368** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11368" target="_blank">Graphite</a> 
                                * **PR #11369** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11369" target="_blank">Graphite</a> 
                                  * **PR #10983** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10983" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/11717?utm_source=stack-comment).

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #11002** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11002" target="_blank">Graphite</a> 
    * **PR #11330** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11330" target="_blank">Graphite</a> 
    * **PR #11001** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11001" target="_blank">Graphite</a> 
    * **PR #11038** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11038" target="_blank">Graphite</a> 
      * **PR #11051** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11051" target="_blank">Graphite</a> 
      * **PR #11362** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11362" target="_blank">Graphite</a> 
        * **PR #11717** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11717" target="_blank">Graphite</a>  👈
          * **PR #11695** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11695" target="_blank">Graphite</a> 
            * **PR #11696** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11696" target="_blank">Graphite</a> 
              * **PR #11697** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11697" target="_blank">Graphite</a> 
                * **PR #11698** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11698" target="_blank">Graphite</a> 
                  * **PR #11699** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11699" target="_blank">Graphite</a> 
                    * **PR #11363** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11363" target="_blank">Graphite</a> 
                      * **PR #11364** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11364" target="_blank">Graphite</a> 
                        * **PR #11365** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11365" target="_blank">Graphite</a> 
                          * **PR #11366** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11366" target="_blank">Graphite</a> 
                            * **PR #11367** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11367" target="_blank">Graphite</a> 
                              * **PR #11368** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11368" target="_blank">Graphite</a> 
                                * **PR #11369** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11369" target="_blank">Graphite</a> 
                                  * **PR #10983** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10983" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/11717?utm_source=stack-comment).

**Review by OwenKephart — DISMISSED** [review]

File: python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py [hunk 1]
```diff
@@ -650,13 +650,16 @@ def determine_asset_partitions_to_reconcile_for_freshness(
             current_data_times = get_current_data_times_for_key(
                 instance_queryer, asset_graph, relevant_upstream_keys, key
             )
+            expected_data_times = get_expected_data_times_for_key(
```
Comment (OwenKephart) [inline python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py:653]: I don't think this line is necessary (if it needs to be there for linting reasons, we should set it to the empty dictionary here)

**Review by OwenKephart — DISMISSED** [review]

**Review by OwenKephart — APPROVED** [review]

**Pull request merged**